libs/entitlement/qa/entitlement-nonmodular/src/main/java/org/elasticsearch/test/entitlements/EntitlementCheckNonModularPlugin.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+package org.elasticsearch.test.entitlements;
+
+import org.elasticsearch.cluster.metadata.IndexNameExpressionResolver;
+import org.elasticsearch.cluster.node.DiscoveryNodes;
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.IndexScopedSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.settings.SettingsFilter;
+import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.features.NodeFeature;
+import org.elasticsearch.plugins.ActionPlugin;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.rest.RestController;
+import org.elasticsearch.rest.RestHandler;
+
+import java.util.List;
+import java.util.function.Predicate;
+import java.util.function.Supplier;
+
+public class EntitlementCheckNonModularPlugin extends Plugin implements ActionPlugin {
+
+    @Override
+    @SuppressForbidden(reason = "Specifically testing System.exit")
+    public List<RestHandler> getRestHandlers(
+        final Settings settings,
+        NamedWriteableRegistry namedWriteableRegistry,
+        final RestController restController,
+        final ClusterSettings clusterSettings,
+        final IndexScopedSettings indexScopedSettings,
+        final SettingsFilter settingsFilter,
+        final IndexNameExpressionResolver indexNameExpressionResolver,
+        final Supplier<DiscoveryNodes> nodesInCluster,
+        Predicate<NodeFeature> clusterSupportsFeature
+    ) {
+        return List.of(new RestEntitlementsCheckSystemExitAction());
```

Comment:
Given we will need to add a lot of these, I wonder if we could pass the method to invoke as a query param, so that we don't need to continue creating new rest endpoints.

Reply:
Perhaps we could default to running all of them?

Reply:
Handlers now have an "action" parameter, and tests use the set of actions known to the handler to probe all of them.